### PR TITLE
Fix int do not match with string after esupgrade

### PR DESCRIPTION
**Pull request by marvin-min**
After ES upgrade, some of the filed type was changed from string to int. Correct the type at ACTV side.

A3 side: https://gitlab.dev.activenetwork.com/activenetwork/a3/merge_requests/42

### CODE REVIEW

**Comment by hlcfan** [comment]
So far, the following clues you can doublecheck
ACTV
evergreen?
display_close_date

A3
showContact
Marvin's story to show article image

File: lib/actv/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module ACTV
-  VERSION = "2.9.0"
+  VERSION = "2.9.1"
```
Comment (RogerTongTong) [inline lib/actv/version.rb]: Please merge with the master. This version number was just used.

File: spec/actv/asset_status_spec.rb [hunk 2]
```diff
@@ -21,4 +21,17 @@
       (asset == other).should be_false
     end
   end
+
+
+  describe '#visible?' do
+    it 'returns true when status assetStatusId is 2' do
+      asset = ACTV::AssetStatus.new(assetStatusId: 2, assetStatusName: "VISIBLE")
+      expect(asset.visible?).to eq true
```
Comment (MarkWang2) [inline spec/actv/asset_status_spec.rb]: to be_true

**Comment by marvin-min** [comment]
@hlcfan  ACTV side done. Please have a look.
@MarkWang2 Thanks for the good style.
@RogerTongTong yes.the version is 2.9.2 now.


Thanks for you guys!!

**Comment by langpingxue** [comment]
A3 side: https://gitlab.dev.activenetwork.com/activenetwork/a3/merge_requests/42

File: lib/actv/asset.rb [hunk 3]
```diff
@@ -214,7 +214,7 @@ def has_location?
     end
 
     def evergreen?
-      self.evergreenAssetFlag.downcase == 'true' rescue false
+      self.evergreenAssetFlag.to_s.downcase == 'true' rescue false
```
Comment (neslom) [inline lib/actv/asset.rb]: if `evergreenAssetFlag` returns a boolean now, can't we just return that value in this method?

```ruby
  def evergreen?
    self.evergreenAssetFlag
  end
```

**Comment by neslom** [comment]
Why are we converting all of these types to strings instead of just checking against the type coming out of a3pi? 

I suggest

```ruby
  id == 2
```

instead of 

```ruby
  id.to_s == '2'
```

**Comment by marvin-min** [comment]
@neslom Thanks for your comments.As you can see there still a lot of assets tagged as 'false' rather false and `assetStatusId` is a string value instead of int value. Here is the example: http://a3pi.active.com/v2/search.json?exclude_children=true&category=articles
This is a safe way to compatible solution to support all cases.

**Comment by neslom** [comment]
@marvin-min Ah that's annoying that there are both types still coming out of a3pi. Okay, this makes sense then. 

I think we're going to see if AS is going to make changes on their side so that the types are the same as they were before the upgrade before merging and releasing this. 

If AS doesn't make changes or we need to release this temporarily as a fix, then I'd say this LGTM. 👍

**Comment by brdrgz** [comment]
Please do not merge this yet, see https://gitlab.dev.activenetwork.com/activenetwork/a3pi/merge_requests/4

**Comment by marvin-min** [comment]
@brdrgz Great!! These had been fixed at A3PI side.

**Pull request closed without merging**